client/src/pages/Login/index.js

```diff
@@ -0,0 +1,148 @@
+import { useReducer, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import {
+  Grid,
+  Typography as T,
+  Inputs as I,
+  Button,
+  TextWithIcon,
+} from '../../components';
+import * as S from './style';
+import validate from '../../validation/schemas/login';
+import { Users } from '../../api-calls';
+
+import { navRoutes as R } from '../../constants';
+const { Row, Col } = Grid;
+
+const initialState = {
+  email: '',
+  password: '',
+  httpError: '',
+  validationErrs: {},
+  submitAttempt: false,
+};
+
+function reducer(state, newState) {
+  return { ...state, ...newState };
+}
+const cleanEmail = (email) => email.toLowerCase().trim();
+
+const Login = () => {
+  const [state, setState] = useReducer(reducer, initialState);
+  const { email, password, validationErrs, httpError, submitAttempt } = state;
+  const history = useHistory();
+  useEffect(() => {
+    if (submitAttempt) {
+      validateForm();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [email, password]);
+
+  const validateForm = () => {
+    try {
+      validate({
+        email: cleanEmail(email),
+        password,
+      });
+      setState({ validationErrs: {} });
+      return true;
+    } catch (error) {
+      if (error.name === 'ValidationError') {
+        setState({ validationErrs: error.inner });
+      }
+      return false;
+    }
+  };
+
+  const handleLogin = async () => {
+    const { error } = await Users.login({
+      email: cleanEmail(email),
+      password,
+    });
+
+    if (error) {
+      if (error.statusCode === 409) {
+        setState({ validationErrs: { email: error.message } });
+      } else {
+        setState({ httpError: error.message });
+      }
+    } else {
+      // after that the user should be directed to its dashboard
+      history.push(R.ADMIN.HOME);
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setState({ submitAttempt: true });
```

Comment:
use `useRef` instead of set state, as this doesn't require to dom re-render

Reply:
done